### PR TITLE
chore!: move blob protos to app under it's own package

### PR DESCRIPTION
**Pull request by cmwaters**
Ref: https://github.com/celestiaorg/celestia-app/issues/2570

This PR creates a `pkg/blob` where the protos for `Blob` and `BlobTx` is created. The intention is that this becomes it's own go.mod to avoid needing to depend on `celestia-core` and our `cosmos-sdk` fork. Other common structures like namespace should do the same so we can achieve some cannonical representation of the structs between node and core/app

The reason for creating the types here instead of in `celestia-core` is so that only `celestia-app` becomes a multi go.mod repo as opposed to either. Additionally, `Blob` and `BlobTx` were never part of the upstream CometBFT.

I also made the decision to have a single representation of blobs rather than a proto and go native. The reason being is one of simplification and performance. With large blobs, copying the data across to another type can lead to a noticeable impact. Having a single struct however does have the dowside of potential coversion errors from uint32 to uint8 so data needs to be validated before it can be used (this is also the same as before).

This PR also ports over the proofs. I have put this all under a common/v1 proto package.

### CODE REVIEW

File: pkg/appconsts/global_consts.go [hunk 1]
```diff
@@ -13,16 +14,16 @@ import (
 // They can not change throughout the lifetime of a network.
 const (
 	// NamespaceVersionSize is the size of a namespace version in bytes.
-	NamespaceVersionSize = 1
+	NamespaceVersionSize = ns.NamespaceVersionSize
 	// NamespaceVersionMaxValue is the maximum value a namespace version can be.
 	// This const must be updated if NamespaceVersionSize is changed.
 	NamespaceVersionMaxValue = math.MaxUint8
```
Comment (rootulp) [inline pkg/appconsts/global_consts.go:20]: [extremely optional] It seems like this constant could also be exported by `pkg/namespace` and imported here

```suggestion
	// NamespaceVersionMaxValue is the maximum value a namespace version can be.
	// This const must be updated if NamespaceVersionSize is changed.
	NamespaceVersionMaxValue = ns.NamespaceVersionMaxValue
```

**Review by rootulp — APPROVED** [review]
👏 this is so clean, great work!

> I also made the decision to have a single representation of blobs rather than a proto and go native

I like this!

[discussion][not blocking] I think it may help contributors if we include a note on why there exists an `x/blob` module and a `pkg/blob` for types. I don't think there's a good place to put this yet but I guess we could add a directory overview section in the README.md. In the absence of that, we could put this note in `x/blob/README.md` or `pkg/blob/README.md` (would need to be created).

**Comment by cmwaters** [comment]
> [discussion][not blocking] I think it may help contributors if we include a note on why there exists an x/blob module and a pkg/blob for types. I don't think there's a good place to put this yet but I guess we could add a directory overview section in the README.md. In the absence of that, we could put this note in x/blob/README.md or pkg/blob/README.md (would need to be created).

Yeah this makes sense. Can follow up in another PR

File: pkg/blob/blob.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	fmt "fmt"
+	math "math"
+	"sort"
+
+	"github.com/celestiaorg/celestia-app/pkg/namespace"
+)
+
+// ProtoBlobTxTypeID is included in each encoded BlobTx to help prevent
+// decoding binaries that are not actually BlobTxs.
+const ProtoBlobTxTypeID = "BLOB"
```
Comment (evan-forbes) [inline pkg/blob/blob.go:15]: [unrelated to this PR, just a comment no change needed]

I know that we need something like this atm, but there _has_ to be better solution to this

**Review by evan-forbes — APPROVED** [review]
I, and probably everyone, really appreciate the refactor to only have a single struct :fire:

File: pkg/blob/blob.go (hunk 2 above)
Comment (evan-forbes) [reply pkg/blob/blob.go:15]: could maybe be solved with a flavor of https://github.com/celestiaorg/celestia-app/issues/1105

**Pull request merged**